sky/clouds/service_catalog/__init__.py

```diff
@@ -71,7 +72,14 @@ def list_accelerators(
               List['common.InstanceTypeInfo']] = collections.defaultdict(list)
     for result in results:
         for gpu, items in result.items():
-            ret[gpu] += items
+            if quantity:
+                new_items = []
+                for item in items:
+                    if item.accelerator_count == quantity:
+                        new_items.append(item)
+                ret[gpu] += new_items
+            else:
+                ret[gpu] += items
```

Comment:
My bad, I should have clarified earlier - we should implement this filtering further downstream in [list_accelerators_impl](https://github.com/skypilot-org/skypilot/blob/9dca02b2c6b0290d46d1ab03829879cee9949d8c/sky/clouds/service_catalog/common.py#L423). That was `results` returned in L67 should already have GPUs filtered by quantity. You may need to add a `quantity_filter` arg to list_accelerators_impl.

Reply:
Pushed all the filtering down into list_accelerators_impl and added the quantity_filter to all the list_accelerator functions in all the cloud catalog files